src/Avalonia.Controls/MenuItemAccessKeyHandler.cs

```diff
@@ -88,9 +88,9 @@ protected virtual void OnTextInput(object? sender, TextInputEventArgs e)
         {
             if (!string.IsNullOrWhiteSpace(e.Text))
             {
-                var text = e.Text.ToUpper();
+                var text = e.Text.ToUpperInvariant();
                 var focus = _registered
-                    .FirstOrDefault(x => x.Item1 == text && x.Item2.IsEffectivelyVisible)?.Item2;
+                    .FirstOrDefault(x => x.Element.IsEffectivelyVisible && x.Item1 == text).Element;
```

Comment:
Why was order changed here?
String comparison is way faster than reading this property: https://github.com/AvaloniaUI/Avalonia/blob/18b1fb74fe7e0393b7768286d01b6e16b50acce9/src/Avalonia.Base/Visual.cs#L192

Reply:
Also, consider string.Equals with Ordinal or Invariant comparison, as by default current culture comparison is used.

Reply:
I inverted the order it is useless to check equality if the MenuIItem is disabled.

Reply:
Yes, but it's way faster to check equality, that check if item is disabled. Check the link, this property also traverses the tree.